homeassistant/components/sonos/switch.py

```diff
@@ -190,18 +190,20 @@ def extra_state_attributes(self):
 
     async def async_turn_on(self, **kwargs) -> None:
         """Turn alarm switch on."""
-        await self.async_handle_switch_on_off(turn_on=True)
+        if not self.alarm.enabled:
```

Comment:
Is it possible that we can have the wrong state here?

Reply:
Assuming that event subscriptions are working properly, this would be a very tight race condition (and I'd say acceptable) if it were being updated from the Sonos app at the same time.

That brings up a separate concern I have about how alarms work when we're in polling mode. I think that needs to be tested more thoroughly.

Reply:
We normally don't stop service calls if we can't guarantee to have the correct state. Otherwise we'd never be able to change state once we have an incorrect state.

Polling integrations fall under this category.

Reply:
Right. Let's just close then as the other change is a tiny optimization that won't matter for a service call.